vignettes/adrs_pcwg3.Rmd

```diff
@@ -0,0 +1,454 @@
+library(pharmaverseadam)
+library(metatools)
+library(dplyr)
+library(tibble)
+library(gt)
+```
+
+
+# Programming Workflow
+  
+-   [Read in Data](#readdata)
+-   [Pre-processing of Input Records](#input)
+-   [Soft Tissue and Bone Lesion Responses](#assessments)
+-   [Bone Lesion Confirmation (`2+2 Flare Rule`)](#boneconfirmation)
+-   [Unconfirmed BOR](#ubor)
+-   [Confirmed BOR](#cbor)
+            
+              
+## Read in Data {#readdata}
+
+To start, all data frames needed for the creation of `ADRS` should be read into the environment. This will be a company specific process. Some of the data frames needed are `ADSL` and `RS`.
+            
+ For example purpose, the SDTM and ADaM datasets (based on CDISC Pilot test data)---which are included in `{pharmaversesdtm}` and `{pharmaverseadam}`---are used.
+              
+The dataset `rs_onco_pcwg3` is expected to already be available, containing soft tissue (`RECIST 1.1`) bone response etc. This vignette assumes the full dataset is preprocessed and ready for analysis.
+            
+```{r message=FALSE}
+adsl <- pharmaverseadam::adsl
+
+# PCWG3 sdtm data
+#rs <- pharmaversesdtm::rs_onco_pcwg3
+
+rs <- rs_onco_pcwg3
+rs <- convert_blanks_to_na(rs)
+```
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  rs,
+   display_vars = exprs(USUBJID, RSTESTCD, RSSTRESC, VISIT,VISITNUM,RSDTC)
+)
+```
+
+At this step, it may be useful to join `ADSL` to your `RS` domain. Only the `ADSL` variables used for derivations are selected 
+at this step.
+
+```{r eval=TRUE}
+adsl_vars <- exprs(RANDDT, TRTSDT)
+adrs <- derive_vars_merged(
+  rs,
+  dataset_add = adsl,
+  new_vars = adsl_vars,
+  by_vars = get_admiral_option("subject_keys")
+)
+```
+
+## Pre-processing of Input Records {#input}
+
+The next step is to assign parameter level values such as PARAMCD, PARAM,PARAMN, PARCAT1, etc. 
+For this, a lookup can be created based on the SDTM RSCAT, RSTESTCD and RSEVAL values to join to the source data.
+
+```{r, eval=TRUE, include=TRUE, message=FALSE}
+param_lookup <- tribble(
+  ~RSCAT, ~RSTESTCD, ~PARAMCD, ~PARAM, ~PARAMN, ~PARCAT1, ~PARCAT1N,
+  
+  # Soft Tissue Response (RECIST 1.1)
+  "RECIST 1.1", "SFTSRESP", "SFTSRESP", "Soft Tissue Response by Investigator", 1, "RECIST 1.1", 1,
+  
+  # Bone Response (PCWG3)
+  "PCWG SCHER PROSTATE CANCER 2016", "BONERESP", "BONERESP", "Bone Response by Investigator", 2, "PCWG3", 2,
+  
+  # Combined Overall Tumor Response
+  "PCWG3 and RECIST 1.1 Modified", "OVRLRESPC", "OVRLRESPC", "Combined Overall Tumor Response by Investigator-Derived", 3, "PCWG3 and RECIST 1.1 Modified", 3,
+  
+  # Tumor Marker Response (PCWG3)
+  "PCWG SCHER PROSTATE CANCER 2016", "TMRESP", "TMRESP", "Tumor Marker Response by Investigator", 4, "PCWG3", 4,
+  
+  # Overall Tumor Response (PCWG3)
+  "PCWG SCHER PROSTATE CANCER 2016", "OVRLRESP", "OVRLRESP", "Overall Tumor Response by Investigator CRF", 5, "PCWG3 Collected", 5
+)
+
+```  
+
+### Table : Overall Time Point Response
+
```

Comment:
Could we add a link to the source of this table? I think it is from https://www.lexjansen.com/pharmasug/2024/DS/PharmaSUG-2024-DS-287.pdf.